### ISSUE TITLE
[BUG] Some key combinations also trigger other key combinations on different keyboard layouts that were not pressed.

### ISSUE THREAD

**Issue by Fyers** [JohannesKlauss/react-hotkeys-hook#909]
**Describe the bug**
Some key combinations also trigger other key combinations on different keyboard layouts that were not pressed.

This behaviour is due to the global event handler in `isHotkeyPressed` that will add both `key` and `code` from the KeyboardEvent instead of only `key`. See Expected behavior.

**To Reproduce**
1. Go to https://codesandbox.io/s/great-blackburn-ksg5nv?file=/src/App.js
2. Open the console
2. Use a keyboard QWERTZ layout instead of QWERTY
3. Press Ctrl+z
4. Undo and redo will be triggered in the console, instead of only one of both. Same for Ctrl+y

**Expected behavior**
`KeyboardEvent.key` should be used on all instances and never together, since `KeyboardEvent.code` is not the characters that will be generated by pressing the key, but the physical key which then will may not reflect the Prompts in the UI. So if my UI tells the user to use "Ctrl+Z" that would meen "Ctrl+Y" on German keyboards which is not expected when using `KeyboardEvent.code`.

**Screenshots**
![image](https://user-images.githubusercontent.com/26491227/212015422-64b5f8ce-900c-42b6-948f-d862f2c80fdd.png)
![image](https://user-images.githubusercontent.com/26491227/212015343-7187faa2-7a73-493d-84a9-21cabaf954bb.png)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
 - Version 108.0.5359.125

**Comment by JohannesKlauss** [comment]
This is an issue I am not sure how to fix.

I used to just check for the `e.key`, but this is not sufficient. Combinations like `shift+1` won't work, because this would result in `e.key === '!'`, not `e.key === '1'` with the shift property set to true.

But since there are tons of keyboard layouts, an app might want to just listen to `shift+1`, no matter the produced key. Checking only for `e.key` cannot fix the issue, since it's highly dependent on the keyboard layout.

**Comment by JohannesKlauss** [comment]
I was thinking that maybe we can introduce something like `useLocalHotkeys` that just listens for the produced key in `e.key` instead of `e.code`. But I am not sure if this is helpful or will cause even more confusion.

**Comment by Fyers** [comment]
Yeah, that's a tough one. I would prefer either `useLocalHotkeys` or an option prop but it needs good documentation with examples when to use what. But it defenetly needs both since it needs to support both use cases.

**Comment by AHorak** [comment]
Have the same problem in my project.
useLocalHotkeys  sounds good.

**Comment by JohannesKlauss** [comment]
Yeah, this will most likely introduce additional complexity. So I'll first have to check on how I should implement this

**Comment by JohannesKlauss** [comment]
This is a layout specific problem that gets addressed in version 5.

This has been fixed in pre release [5.0.0-1](https://www.npmjs.com/package/react-hotkeys-hook/v/5.0.0-1). Please try out this version.
To listen to `z` no matter the keyboard layout use you just use `useHotkeys('z', callback, {useKey: true})`

Please let me know if that fixes your issue.
